libraries/Microsoft.Bot.Builder.LanguageGeneration/LGFile.cs

```diff
@@ -0,0 +1,385 @@
+
+        /// <summary>
+        /// Gets get all diagnostics from current lg file and reference lg files.
+        /// </summary>
+        /// <value>
+        /// All diagnostics from current lg file and reference lg files.
+        /// </value>
+        public IList<Diagnostic> AllDiagnostics => new List<LGFile> { this }.Union(References).SelectMany(x => x.Diagnostics).ToList();
+
+        /// <summary>
+        /// Gets or sets delegate for resolving resource id of imported lg file.
+        /// </summary>
+        /// <value>
+        /// Delegate for resolving resource id of imported lg file.
+        /// </value>
+        public ImportResolverDelegate ImportResolver { get; set; }
+
+        /// <summary>
+        /// Gets or sets templates that this LG file contains directly.
+        /// </summary>
+        /// <value>
+        /// templates that this LG file contains directly.
+        /// </value>
+        public IList<LGTemplate> Templates { get; set; }
+
+        /// <summary>
+        /// Gets or sets expression parser.
+        /// </summary>
+        /// <value>
+        /// expression parser.
+        /// </value>
+        public ExpressionEngine ExpressionEngine { get; set; }
+
+        /// <summary>
+        /// Gets or sets import elements that this LG file contains directly.
+        /// </summary>
+        /// <value>
+        /// import elements that this LG file contains directly.
+        /// </value>
+        public IList<LGImport> Imports { get; set; }
+
+        /// <summary>
+        /// Gets or sets all references that this LG file has from <see cref="Imports"/>.
+        /// Notice: reference includs all child imports from the lg file,
+        /// not only the children belong to this lgfile directly.
+        /// so, reference count may >= imports count. 
+        /// </summary>
+        /// <value>
+        /// all references that this LG file has from <see cref="Imports"/>.
+        /// </value>
+        public IList<LGFile> References { get; set; }
+
+        /// <summary>
+        /// Gets or sets diagnostics.
+        /// </summary>
+        /// <value>
+        /// diagnostics.
+        /// </value>
+        public IList<Diagnostic> Diagnostics { get; set; }
+
+        /// <summary>
+        /// Gets or sets LG content.
+        /// </summary>
+        /// <value>
+        /// LG content.
+        /// </value>
+        public string Content { get; set; }
+
+        /// <summary>
+        /// Gets or sets id of this LG file.
+        /// </summary>
+        /// <value>
+        /// id of this lg source. For file, is full path.
+        /// </value>
+        public string Id { get; set; }
+
+        /// <summary>
+        /// use an existing LGFile to override current object.
+        /// </summary>
+        /// <param name="lgFile">Existing LGFile.</param>
+        public void Initialization(LGFile lgFile)
```

Comment:
Init or Initialize would be good enough, Initialization is a noun, not suitable for a method name